### DISCUSSION TITLE
Drop Down Menu Problem mit React Select

### DISCUSSION THREAD

**Discussion by JanaS87** [neuefische/questions#676] [Web] [answered]
Hallo zusammen,

aktuell arbeite ich an meiner Katzenfutter App und habe in US3 die "add new cat" page erstellt.
Dort soll mithilfe der Drop Down Menüs das Katzenfutter ausgewählt und hinzugefügt werden.

Die Auflistung der Futter- Sorten funktioniert, die Suche im Auswahl- Fenster funktioniert ebenfalls. 

Der Fehler:

Sobald ein Futter ausgewählt wird, erscheint es nicht im Auswahlfenster, das Hinzufügen mit dem entsprechenden Button funktioniert ebenfalls nicht. Interessanterweise wird auch der Placeholder nicht angezeigt.
Ich habe den Code schon durch ChatGPT gejagt, bei Google gesucht, in der Documentation von React Select gestöbert aber leider nichts gefunden, was annähernd auf dieses Problem passt.

![drop down problem](https://github.com/neuefische/questions/assets/136725332/16b675e2-10a3-4882-af0f-8881a1823d9c)

Jedes Console.log() zeigt, dass die Daten empfangen und überreicht werden aber sie kommen nicht an.

Es gibt mehrere Vermutungen, es könnte am handleAddGoodFood() / handleAddBadFood() liegen oder am handleChange(event).
![handleAddFood](https://github.com/neuefische/questions/assets/136725332/97383065-ce87-4e11-a2c8-0b80eb091745)
![handleChange](https://github.com/neuefische/questions/assets/136725332/bfca8a71-d9d4-45a2-b6cd-7a8444080071)

Möglicherweise auch an den handleGoodFoodChange().
![handleFoodChange](https://github.com/neuefische/questions/assets/136725332/e2e18cf1-cf71-451a-a59e-fd8192a524ee)

Es könnte aber auch im Select selbst liegen oder aber an den styled components, dass da vielleicht irgendwo was zwischen haut.
![Select](https://github.com/neuefische/questions/assets/136725332/76ef908f-a872-4c16-8cf9-c69fbdbd81f5)

Zu der Page kommt ihr, wenn ihr auf Add new Cat klickt.

Ich habe dafür einen PR erstellt, welcher hier zu finden ist:
https://github.com/JanaS87/capstone-project/pull/8

Über jegliche Tips, Hilfe, Lösungen freue ich mich sehr! 
Viele Grüße
Jana

**Comment by Kheirah** [comment, marked as answer]
In your `styles.js` there's one single line that's responsible for this behavior:
```js
export default createGlobalStyle`
  *,
  *::before,
  *::after {
    box-sizing: border-box;
    background-color:#F3E99F; //this line has to go bye bye
  }
...
```
With this line you said, I want to have a yellow background on _ALL ELEMENTS._ And you got it 😄  Even the `select` element got it, which is the reason why the selected value was _hiding behind that yellow background._ Remove it and the selected element will actually appear selected.

In case it doesn't work yet, this is what the `Select` looks like for me:
```js
<Select
  options={catFoodOptions}
  defaultValue={catFoodOptions[0]}
  onChange={handleGoodFoodChange}
  placeholder="choose food..." 
/>
```
PS:
English, please =)
